![1*NDQR0S6ZkD4Hb1h1QIUdFQ.png](https://cdn-images-1.medium.com/max/1600/1*NDQR0S6ZkD4Hb1h1QIUdFQ.png  )

## Table of Contents

1. [Problem Statement](#section1)<br>
2. [DataSet Loading and Brif Description](#section2)<br>
    - 2.1 [Variable Notes](#section21)<br>
    - 2.2 [Data Import](#section22)<br>
3. [Examining Data Available From An Existing Information Source-Data profiling](#section3)<br>
    - 3.1 [Basic DataSet Understanding](#section31)<br>
    - 3.2 [Pre-Profile](#section32)<br>
    - 3.3 [Data Preprocessing](#section33)<br>
        - 3.3.1 [Age NULL Value Replace with Median](#section331)<br>
        - 3.3.2 [](#section332)<br>
    - 3.4 [Post  Profiling](#section34)<br>
4. [Observation & Questions](#section4)<br>

# 1. Problem Statment

Movies lots of metadata about movies.... So actually what is deciding success or failure of movie? That is we only we are going to find out.
- What is it really that describes their success or failure? 
- Revenu at box office?
- success or failure association of popular directors or actors?
- success or failure movie promotion ?

So basically What makes movies good or bad? Here i got opportunity to find few question.

# 2. DataSet Loading and Brif Description

Data set of It contained data from 5043 movies spread across 28 features. The data points included are:


|Column Name|Description|
|---------------------------|---------------------------|
|movie_title| Contains title of a movie|
|color| Specifies whether a movie is black and white or color|
|num_critic_for_reviews| Contains number of critic reviews per movie|
|movie_facebook_likes| Contains number of facebook likes per movie|
|duration| Contains duration of a movie in minutes|
|director_name| Contains name of the director of a movie|
|director_facebook_likes| Contains number of facebook likes for a director|
|actor_3_name| Contains the name of the 3rd leading actor of a movie|
|actor_3_facebook_likes| Contains number of facebook likes for actor 3|
|actor_2_name| Contains name of 2nd leading actor of a movie|
|actor_2_facebook_likes| Contains number of facebook likes for actor 2|
|actor_1_name| Contains name of the actor in lead role|
|actor_1_facebook_likes| Contains number of facebook likes for actor 1|
|gross| Contains the amount a movie grossed in USD|
|genres| Contains the sub-genres to which a movie belongs|
|num_voted_users| Contains number of users votes for a movie|
|cast_total_facebook_likes| Contains number of facebook likes for the entire cast of a movie|
|facenumber_in_poster| Contains number of actor’s faces on a movie poster|
|plot_keywords| Contains key plot words associated with a movie|
|movie_imdb_link| Contains the link to the imdb movie page|
|num_user_for_reviews| Contains the number of user generated reviews per movie|
|language| Contains the language of a movie|
|country| Contains the name of the country in which a movie was made|
|content_rating| Contains maturity rating of a movie|
|budget| Contains the amount of money spent in production per movie (not always in USD)|
|title_year| Contains the year in which a film was released|
|imdb_score| Contains user generated rating per movie|
|aspect_ratio| Contains the size of the aspect ratio of a movie|


## 2.1 Varaible Note



## 2.2 Data Load

In [1]:
# Importing packges

import numpy as np              # milti-dimensional array and matrices
import pandas as pd             # manipulation and analysis
import pandas_profiling 
import matplotlib.pyplot as plt # umerical mathematics extension NumPy ( Ploting libarary)
import seaborn as sns           # Provides a high level interface for drawing attractive and informative statistical graphics


'''
%matplotlib inline sets the backend of matplotlib to the 'inline' backend: With this backend, the output of 
plotting commands is displayed inline within frontends like the Jupyter notebook, directly below the code cell that
produced it.
'''

%matplotlib inline

In [67]:
# we can see the value of multiple statements at once
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"
MoviMetaData = pd.read_csv("C://Users//t.shah//Desktop//TanujGit//ExploratoryDataAnalysis-EDA//DataSet/movie_metadata.csv")
MoviMetaData.head()
MoviMetaData.tail()

color      director_name  num_critic_for_reviews   duration  \
0  Color      James Cameron              723.000000 178.000000   
1  Color     Gore Verbinski              302.000000 169.000000   
2  Color         Sam Mendes              602.000000 148.000000   
3  Color  Christopher Nolan              813.000000 164.000000   
4    NaN        Doug Walker                     nan        nan   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                 0.000000              855.000000  Joel David Moore   
1               563.000000             1000.000000     Orlando Bloom   
2                 0.000000              161.000000      Rory Kinnear   
3             22000.000000            23000.000000    Christian Bale   
4               131.000000                     nan        Rob Walker   

   actor_1_facebook_likes            gross                           genres  \
0             1000.000000 760505847.000000  Action|Adventure|Fantasy|Sci-Fi   
1            40000.000000 309404152.000000         Action|Adventure|Fantasy   
2            11000.000000 200074175.000000        Action|Adventure|Thriller   
3            27000.000000 448130642.000000                  Action|Thriller   
4              131.000000              nan                      Documentary   

   ... num_user_for_reviews language  country  content_rating  \
0  ...          3054.000000  English      USA           PG-13   
1  ...          1238.000000  English      USA           PG-13   
2  ...           994.000000  English       UK           PG-13   
3  ...          2701.000000  English      USA           PG-13   
4  ...                  nan      NaN      NaN             NaN   

            budget  title_year actor_2_facebook_likes imdb_score  \
0 237000000.000000 2009.000000             936.000000   7.900000   
1 300000000.000000 2007.000000            5000.000000   7.100000   
2 245000000.000000 2015.000000             393.000000   6.800000   
3 250000000.000000 2012.000000           23000.000000   8.500000   
4              nan         nan              12.000000   7.100000   

   aspect_ratio movie_facebook_likes  
0      1.780000                33000  
1      2.350000                    0  
2      2.350000                85000  
3      2.350000               164000  
4           nan                    0  

[5 rows x 28 columns]

color     director_name  num_critic_for_reviews   duration  \
5038  Color       Scott Smith                1.000000  87.000000   
5039  Color               NaN               43.000000  43.000000   
5040  Color  Benjamin Roberds               13.000000  76.000000   
5041  Color       Daniel Hsia               14.000000 100.000000   
5042  Color          Jon Gunn               43.000000  90.000000   

      director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
5038                 2.000000              318.000000     Daphne Zuniga   
5039                      nan              319.000000     Valorie Curry   
5040                 0.000000                0.000000     Maxwell Moody   
5041                 0.000000              489.000000     Daniel Henney   
5042                16.000000               16.000000  Brian Herzlinger   

      actor_1_facebook_likes        gross                        genres  ...  \
5038              637.000000          nan                  Comedy|Drama  ...   
5039              841.000000          nan  Crime|Drama|Mystery|Thriller  ...   
5040                0.000000          nan         Drama|Horror|Thriller  ...   
5041              946.000000 10443.000000          Comedy|Drama|Romance  ...   
5042               86.000000 85222.000000                   Documentary  ...   

     num_user_for_reviews language  country  content_rating      budget  \
5038             6.000000  English   Canada             NaN         nan   
5039           359.000000  English      USA           TV-14         nan   
5040             3.000000  English      USA             NaN 1400.000000   
5041             9.000000  English      USA           PG-13         nan   
5042            84.000000  English      USA              PG 1100.000000   

      title_year actor_2_facebook_likes imdb_score  aspect_ratio  \
5038 2013.000000             470.000000   7.700000           nan   
5039         nan             593.000000   7.500000     16.000000   
5040 2013.000000               0.000000   6.300000           nan   
5041 2012.000000             719.000000   6.300000      2.350000   
5042 2004.000000              23.000000   6.600000      1.850000   

     movie_facebook_likes  
5038                   84  
5039                32000  
5040                   16  
5041                  660  
5042                  456  

[5 rows x 28 columns]

# 3.Examining Data Available From An Existing Information Source-Data profiling

## 3.1 Basic DataSet Understanding

In [4]:
MoviMetaData.shape

(5043, 28)

In [5]:
MoviMetaData.columns

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')

In [6]:
MoviMetaData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5043 entries, 0 to 5042
Data columns (total 28 columns):
color                        5024 non-null object
director_name                4939 non-null object
num_critic_for_reviews       4993 non-null float64
duration                     5028 non-null float64
director_facebook_likes      4939 non-null float64
actor_3_facebook_likes       5020 non-null float64
actor_2_name                 5030 non-null object
actor_1_facebook_likes       5036 non-null float64
gross                        4159 non-null float64
genres                       5043 non-null object
actor_1_name                 5036 non-null object
movie_title                  5043 non-null object
num_voted_users              5043 non-null int64
cast_total_facebook_likes    5043 non-null int64
actor_3_name                 5020 non-null object
facenumber_in_poster         5030 non-null float64
plot_keywords                4890 non-null object
movie_imdb_link              5043 non-

#### Observation:


- Total row __5043 __ and columns are __28__
- Most of columns (Values ) has null values.


In [21]:
total = MoviMetaData.isnull().sum().sort_values(ascending=False)
percent_1 = MoviMetaData.isnull().sum()/MoviMetaData.isnull().count()*100
percent_2 = (round(percent_1, 1))
missing_data = pd.concat([total, percent_2], axis=1, keys=['Total', '%'])
missing_data.sort_values(by = '%', ascending=True).head(30)

C:\Users\t.shah\AppData\Local\Continuum\anaconda3\lib\site-packages\ipykernel_launcher.py:4: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  after removing the cwd from sys.path.


Total     %
movie_facebook_likes           0   0.0
genres                         0   0.0
movie_imdb_link                0   0.0
movie_title                    0   0.0
num_voted_users                0   0.0
cast_total_facebook_likes      0   0.0
imdb_score                     0   0.0
actor_1_facebook_likes         7   0.1
actor_1_name                   7   0.1
country                        5   0.1
language                      12   0.2
duration                      15   0.3
facenumber_in_poster          13   0.3
actor_2_name                  13   0.3
actor_2_facebook_likes        13   0.3
num_user_for_reviews          21   0.4
color                         19   0.4
actor_3_name                  23   0.5
actor_3_facebook_likes        23   0.5
num_critic_for_reviews        50   1.0
director_name                104   2.1
director_facebook_likes      104   2.1
title_year                   108   2.1
plot_keywords                153   3.0
content_rating               303   6.0
aspect_ratio                 329   6.5
budget                       492   9.8
gross                        884  17.5

In [72]:
pd.set_option('float_format', '{:f}'.format)
pd.set_option('display.max_columns', 100) 
MoviMetaData.describe(include=[np.number]) # have taken only Number Datatypes

num_critic_for_reviews    duration  director_facebook_likes  \
count             4949.000000 4983.000000              4895.000000   
mean               139.890079  107.213325               688.679060   
std                121.477586   25.248775              2821.649616   
min                  1.000000    7.000000                 0.000000   
25%                 50.000000   93.000000                 7.000000   
50%                110.000000  103.000000                49.000000   
75%                195.000000  118.000000               192.000000   
max                813.000000  511.000000             23000.000000   

       actor_3_facebook_likes  actor_1_facebook_likes            gross  \
count             4975.000000             4991.000000      4124.000000   
mean               639.900905             6556.939892  48325648.887488   
std               1643.298282            15061.586700  67964826.775601   
min                  0.000000                0.000000       162.000000   
25%                133.000000              611.500000   5304835.250000   
50%                369.000000              984.000000  25445749.000000   
75%                635.000000            11000.000000  62319416.000000   
max              23000.000000           640000.000000 760505847.000000   

       num_voted_users  cast_total_facebook_likes  facenumber_in_poster  \
count      4998.000000                4998.000000           4985.000000   
mean      83470.198679                9676.941176              1.368907   
std      138086.560369               18165.404578              2.014623   
min           5.000000                   0.000000              0.000000   
25%        8560.000000                1405.500000              0.000000   
50%       34260.500000                3085.500000              1.000000   
75%       96120.750000               13740.500000              2.000000   
max     1689764.000000              656730.000000             43.000000   

       num_user_for_reviews             budget  title_year  \
count           4977.000000        4511.000000 4891.000000   
mean             272.014667    39747870.007980 2002.468820   
std              377.776210   206968901.477187   12.475235   
min                1.000000         218.000000 1916.000000   
25%               64.000000     6000000.000000 1999.000000   
50%              156.000000    20000000.000000 2005.000000   
75%              324.000000    45000000.000000 2011.000000   
max             5060.000000 12215500000.000000 2016.000000   

       actor_2_facebook_likes  imdb_score  aspect_ratio  movie_facebook_likes  
count             4985.000000 4998.000000   4671.000000           4998.000000  
mean              1642.998796    6.441056      2.221417           7487.430172  
std               4030.925303    1.124107      1.391185          19290.726563  
min                  0.000000    1.600000      1.180000              0.000000  
25%                280.000000    5.800000      1.850000              0.000000  
50%                595.000000    6.600000      2.350000            162.500000  
75%                917.000000    7.200000      2.350000           3000.000000  
max             137000.000000    9.500000     16.000000         349000.000000

### Observation :
- The mean, min and max are self-explanatory.
- std is the standard deviation, which shows how data is dispersed.


In [24]:
MoviMetaData.select_dtypes(include=['object']).head()

color      director_name      actor_2_name  \
0  Color      James Cameron  Joel David Moore   
1  Color     Gore Verbinski     Orlando Bloom   
2  Color         Sam Mendes      Rory Kinnear   
3  Color  Christopher Nolan    Christian Bale   
4    NaN        Doug Walker        Rob Walker   

                            genres     actor_1_name  \
0  Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1         Action|Adventure|Fantasy      Johnny Depp   
2        Action|Adventure|Thriller  Christoph Waltz   
3                  Action|Thriller        Tom Hardy   
4                      Documentary      Doug Walker   

                                         movie_title          actor_3_name  \
0                                            Avatar              Wes Studi   
1          Pirates of the Caribbean: At World's End         Jack Davenport   
2                                           Spectre       Stephanie Sigman   
3                             The Dark Knight Rises   Joseph Gordon-Levitt   
4  Star Wars: Episode VII - The Force Awakens    ...                   NaN   

                                       plot_keywords  \
0             avatar|future|marine|native|paraplegic   
1  goddess|marriage ceremony|marriage proposal|pi...   
2                bomb|espionage|sequel|spy|terrorist   
3  deception|imprisonment|lawlessness|police offi...   
4                                                NaN   

                                     movie_imdb_link language country  \
0  http://www.imdb.com/title/tt0499549/?ref_=fn_t...  English     USA   
1  http://www.imdb.com/title/tt0449088/?ref_=fn_t...  English     USA   
2  http://www.imdb.com/title/tt2379713/?ref_=fn_t...  English      UK   
3  http://www.imdb.com/title/tt1345836/?ref_=fn_t...  English     USA   
4  http://www.imdb.com/title/tt5289954/?ref_=fn_t...      NaN     NaN   

  content_rating  
0          PG-13  
1          PG-13  
2          PG-13  
3          PG-13  
4            NaN

## 3.2 Pre-Profile

In [28]:
profile = pandas_profiling.ProfileReport(MoviMetaData)
profile.to_file(outputfile="MoviMetaData_Preprocessing.html")

## 3.3 Data Preprocesing Observation

__Details Onservation :__

|Column_Name |Description|Remarks|
|--------------------------|--------------------------|--------------------------|
|actor_1_name | High cardinality| 2098 distinct values |
|actor_2_facebook_likes | 55 / 1.1% zeros Zeros||
|actor_2_name | High cardinality| 3033 distinct values |
|actor_3_facebook_likes | 89 / 1.8% zeros Zeros||
|actor_3_name | High cardinality| 3522 distinct values |
|aspect_ratio | 329 / 6.5% missing values Missing||
|budget| is highly skewed (γ1 = 48.157) Skewed||
|budget | 492 / 9.8% missing values Missing||
|cast_total_facebook_likes| is highly correlated with actor_1_facebook_likes (ρ = 0.95166)||
|content_rating | 303 / 6.0% missing values Missing||
|country | High cardinality| 66 distinct values |
|director_facebook_likes | 907 / 18.0% zeros Zeros||
|director_facebook_likes | 104 / 2.1% missing values Missing||
|director_name | 104 / 2.1% missing values Missing||
|director_name | High cardinality| 2399 distinct values |
|facenumber_in_poster | 2152 / 42.7% zeros Zeros||
|genres | High cardinality| 914 distinct values |
|gross | 884 / 17.5% missing values Missing||
|movie_facebook_likes | 2181 / 43.2% zeros Zeros||
|movie_imdb_link | High cardinality| 4919 distinct values |
|movie_title | High cardinality| 4917 distinct values |
|plot_keywords | 153 / 3.0% missing values Missing||
|plot_keywords | High cardinality| 4761 distinct values |
|title_year | 108 / 2.1% missing values Missing||
|Dataset | 45 duplicate rows ||


THis notbook we are gone work on :

|Column_Name |description|
|--------------------------|--------------------------|
|actor_2_facebook_likes | 55 / 1.1% zeros Zeros|
|actor_3_facebook_likes | 89 / 1.8% zeros Zeros|
|aspect_ratio | 329 / 6.5% missing values Missing|
|budget | 492 / 9.8% missing values Missing|
|content_rating | 303 / 6.0% missing values Missing|
|director_facebook_likes | 907 / 18.0% zeros Zeros|
|director_facebook_likes | 104 / 2.1% missing values Missing|
|director_name | 104 / 2.1% missing values Missing|
|facenumber_in_poster | 2152 / 42.7% zeros Zeros|
|gross | 884 / 17.5% missing values Missing|
|movie_facebook_likes | 2181 / 43.2% zeros Zeros|
|plot_keywords | 153 / 3.0% missing values Missing|
|title_year | 108 / 2.1% missing values Missing|
|Dataset | 45 duplicate rows |



## 3.3 Working on Pre-Proessing Observation

### 3.3.1 Drop Duplicates Records

In [68]:
MoviMetaData = MoviMetaData.drop_duplicates()

In [69]:
MoviMetaData[MoviMetaData['movie_title'].isnull()]

Empty DataFrame
Columns: [color, director_name, num_critic_for_reviews, duration, director_facebook_likes, actor_3_facebook_likes, actor_2_name, actor_1_facebook_likes, gross, genres, actor_1_name, movie_title, num_voted_users, cast_total_facebook_likes, actor_3_name, facenumber_in_poster, plot_keywords, movie_imdb_link, num_user_for_reviews, language, country, content_rating, budget, title_year, actor_2_facebook_likes, imdb_score, aspect_ratio, movie_facebook_likes]
Index: []

[0 rows x 28 columns]

In [70]:
MoviMetaData[MoviMetaData['movie_title'] .isnull()]

Empty DataFrame
Columns: [color, director_name, num_critic_for_reviews, duration, director_facebook_likes, actor_3_facebook_likes, actor_2_name, actor_1_facebook_likes, gross, genres, actor_1_name, movie_title, num_voted_users, cast_total_facebook_likes, actor_3_name, facenumber_in_poster, plot_keywords, movie_imdb_link, num_user_for_reviews, language, country, content_rating, budget, title_year, actor_2_facebook_likes, imdb_score, aspect_ratio, movie_facebook_likes]
Index: []

[0 rows x 28 columns]

### 3.3.1 Title Year Replace with Median

In [77]:
Median_Year = MoviMetaData.title_year.median()
MoviMetaData.title_year.fillna(Median_Year, inplace = True)

In [79]:
MoviMetaData[MoviMetaData['title_year'].isnull()]

Empty DataFrame
Columns: [color, director_name, num_critic_for_reviews, duration, director_facebook_likes, actor_3_facebook_likes, actor_2_name, actor_1_facebook_likes, gross, genres, actor_1_name, movie_title, num_voted_users, cast_total_facebook_likes, actor_3_name, facenumber_in_poster, plot_keywords, movie_imdb_link, num_user_for_reviews, language, country, content_rating, budget, title_year, actor_2_facebook_likes, imdb_score, aspect_ratio, movie_facebook_likes]
Index: []